src/Processors/QueryPlan/Optimizations/optimizeDirectReadFromTextIndex.cpp

```diff
@@ -101,24 +101,28 @@ class FullTextMatchingFunctionDAGReplacer
     /// Replaces text-search functions by virtual columns.
     /// Example: hasToken(text_col, 'token') -> __text_index_text_col_idx_hasToken_0
     /// Returns a pair of (added columns by index name, removed columns)
-    std::pair<IndexReadColumns, Names> replace()
+    std::tuple<IndexReadColumns, Names, String> replace(const String & origin_prewhere_column_name)
```

Comment:
The argument passed to this function is not always the prewhere column. In the fist call it is the filter column, which is totally valid to be non-index column right?